File: ru.kmorozov.library.data.model/src/main/java/ru/kmorozov/library/spring/LibraryContext.java
package ru.kmorozov.library.spring;

import org.springframework.context.support.StaticApplicationContext;
import ru.kmorozov.library.data.repository.BooksRepository;

/**
 * Created by sbt-morozov-kv on 15.12.2016.
 */
public final class LibraryContext extends StaticApplicationContext {

    public static final LibraryContext LIBRARY_CONTEXT = new LibraryContext();

    private LibraryContext() {
        this.registerSingleton("booksRepository", BooksRepository.class);
    }
}

File: ru.kmorozov.library.data.model/src/main/java/ru/kmorozov/library/spring/LibraryContext.kt
package ru.kmorozov.library.spring

import org.springframework.context.support.StaticApplicationContext
import ru.kmorozov.library.data.repository.BooksRepository

/**
 * Created by sbt-morozov-kv on 15.12.2016.
 */
class LibraryContext private constructor() : StaticApplicationContext() {

    init {
        registerSingleton("booksRepository", BooksRepository::class.java)
    }

    companion object {

        val LIBRARY_CONTEXT = LibraryContext()
    }
}
